### PR TITLE
basic UI for school library

### PR DESCRIPTION
**Pull request by levy002**
Build interactive console app.

- [x]  Create  `app.rb` file that will serve as our console app entry-point. It has methods that do the following:

> List all books.

> List all people.

> Create a person (teacher or student, not a plain Person).

> Create a book.

> Create a rental.

> List all rentals for a given person `id`.

- [x]  In `main.rb` define the entry point, this will be a method called main that is invoked at the end of the file. This method does the following:

> Present the user with a list of options to perform.

> Let users choose an option.

> If needed, ask for parameters for the option.

> Have a way to quit the app.

Cheers!!

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  end
+
+  def books_list
+    puts 'No books available!!' if @books.empty?
+    puts "There are #{@books.count} books available"
+
+    @books.each_with_index do |book, index|
+      puts "#{index + 1}) Title: \"#{book.title}\" | Author: #{book.author}"
+    end
+  end
```
Comment (julie-ify) [inline app.rb]: - [x] Your app exhibits unexpected behavior. It quits after I try to list a book. According to the video shared with you, after an option has been processed, the app should still be running and present the user with the list of choices again. Kindly fix this.

  levy-1

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    when 'n'
+      student = Student.new(age, classroom, name, parent_permission: false)
+    else
+      puts 'Invalid option, please try again'
+      return
+    end
+    @people << student
+    puts ''
+    puts "Student is created successfully.Student ID is #{student.id}"
+  end
```
Comment (julie-ify) [inline app.rb]: - [x] The app crashes and quits when I try to create a student. Kindly fix that.

  levy-2

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+  def create_teacher(age, name, specialization)
+    teacher = Teacher.new(age, name, specialization)
+    @people << teacher
+    puts ''
+    puts "Teacher created successfully. Teacher ID is #{teacher.id}"
+  end
```
Comment (julie-ify) [inline app.rb]: - [x] The app quits after creating a teacher but the app should have been running and presented the user with the list of choices again. Kindly fix this behavior.

  
  levy-3

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    if title.strip != '' && author.strip != ''
+      book = Book.new(title, author)
+      @books << book
+      puts ''
+      puts 'Book created successfully'
+    else
+      puts ''
+      puts 'Enter book title and author'
+    end
+  end
```
Comment (julie-ify) [inline app.rb]: - [x] The app quits after creating a book. Kindly fix that.

  
  levy-4

File: main.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  when '2'
+    peoples_list
+  when '3'
+    create_person
+  when '4'
+    create_book
+  when '5'
+    new_rental
+  when '6'
+    rentals_list
```
Comment (julie-ify) [inline main.rb]: - [x] Your entire app does not work as expected. After a user option has been processed, this list should be present again, but that is not the case. Kindly fix that.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @levy002,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KDlamini — APPROVED** [review]
### Hi @levy002 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes.  Your UI works as expected 👏🏾 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!🎊 🥳 

## Highlights 🌞 
✔️  The student's school library console app behaves correctly
✔️  Correct PR title and description
✔️  Correct use of Gitflow 
✔️  No linter errors

# Status: Approved ✔️ 

giphyapproved
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**